src/js_native_api.h

```diff
@@ -370,6 +374,9 @@ NAPI_EXTERN napi_status napi_throw_type_error(napi_env env,
 NAPI_EXTERN napi_status napi_throw_range_error(napi_env env,
                                          const char* code,
                                          const char* msg);
+NAPI_EXTERN napi_status napi_throw_syntax_error(napi_env env,
```

Comment:
This is a new api and it should be enabled / disabled throw `NAPI_EXPERIMENTAL`:
```cpp
#ifdef NAPI_EXPERIMENTAL
NAPI_EXTERN 
napi_status napi_throw_syntax_error(napi_env env,
                                    const char* code,
                                    const char* msg,);
#endif  // NAPI_EXPERIMENTAL
```

Reply:
sure, it's here: https://github.com/nodejs/node/pull/40736/commits/6fafc065898505f02c426da1885d76bed105c627

Reply:
Sorry my mistake I didn't see it.

Reply:
@NickNaso I see that adding this `ifdef` failed the tests on Mac (due to `implicit declaration of function 'napi_throw_syntax_error' is invalid in C99 [-Werror,-Wimplicit-function-declaration]`), tho not on the other platforms.

I assume it's either that the compiler is stricter on Mac (or defined to be stricter), or that the tests in this setup are not running with `NAPI_EXPERIMENTAL`.

What do you recommend to do?

Reply:
You should add this declaration:
```cpp
#define NAPI_EXPERIMENTAL
```
to the head of file `/test/js-native-api/test_error/test_error.c`

Reply:
@NickNaso Thanks. It helped.

The failing action (`test-asan / test-asan (pull_request)`) seems flaky, I see:

```
[UNEXPECTED_FAILURE][FAIL] Entry 0 startTime is approximately correct (up to 20ms difference allowed)
assert_approx_equals: expected 4102.578573999926 +/- 20 but got 4077.9045040002093
    at Test.<anonymous> (/home/runner/work/node/node/test/fixtures/wpt/user-timing/mark.any.js:61:8)
    at Test.step (/home/runner/work/node/node/test/fixtures/wpt/resources/testharness.js:2087:25)
    at test (/home/runner/work/node/node/test/fixtures/wpt/resources/testharness.js:557:30)
    at test_mark (/home/runner/work/node/node/test/fixtures/wpt/user-timing/mark.any.js:59:4)
    at /home/runner/work/node/node/test/fixtures/wpt/user-timing/mark.any.js:117:3
Command: /home/runner/work/node/node/out/Release/node  /home/runner/work/node/node/test/wpt/test-user-timing.js mark.any.js
```

I don't have the permissions to re-run. can you please?

Reply:
> @NickNaso yes, that is what we agreed to do for new functions use node_api instead of napi
> 
> It is what we said we'd do in https://nodejs.medium.com/renaming-n-api-to-node-api-27aa8ca30ed8

Please @idan-at could you please rename the functions you added like reported below?
- node_api_throw_syntax_error
- node_api_create_syntax_error

Reply:
of course - did it here https://github.com/nodejs/node/pull/40736/commits/a9cc563b06edf40aa17798035460d5e44444641d